docs/guides/development/data-synchronization.md

```diff
@@ -0,0 +1,169 @@
+
+When you have [created your Voucherify account](doc:getting-started), you can import your existing data and then synchronize it for future use in campaigns.
+
+> 📘 **Goals**:
+>
+> - Learn why data synchronization is important
+> - Learn how to synchronize your data through data import and updates
+
+> 👍 **Outcome**:
+> 
+> Your data is synchronized with Voucherify.
+
+## Customer synchronization
+
+Voucherify can store and act on collected customer data. The most common customer data operation is to create customer segments based on customer attributes. Customer segments can be used to understand customer behavior, build promotion limits, or trigger automatic incentive delivery.
+
+Voucherify uses the [customer object](ref:customer-object) for [validation](ref:validation-object), [redemption](ref:redemption-object), and [distribution](https://support.voucherify.io/article/19-how-does-the-distribution-manager-work "How does the distribution manager work?") purposes.
+
+The customer data does not have to be stored in Voucherify before a validation or redemption request is made. Voucherify can use the customer data from the request to create a new customer with a unique `id` field.
+
+However, the customer data must exist in Voucherify before a distribution is made. Also, distribution can be made on the basis of the customer's attributes and sent out to customers from a given segment. If the customer data is not up to date, the distribution messages sent manually can reach the wrong audience.
+
+<!-- this bit must be moved someplace
+
+Every customer has unique identification data in the `source_id` field. The `source_id` can be a customer ID or email from a CRM system, database, or a third-party service.It can be imported into Voucherify from your [CRM database]() or a CSV file <!-- link when ready! -->
+
+> 🚧 Customer data synchronization and `source_id`
+>
+> If your data is to be synchronized based on the `source_id` of the customer, you need to import the `source_id` when the customer is uploaded for the first time. You will not be able to update `source_id` later on.
+
+### Customer synchronization options
+
+<!-- https://success.voucherify.io/article/430-crm-playbook
+Adding customers
+
+These are the ways you can add customer profiles to Voucherify:
+
+API
+Webhooks
+The import with a CSV file. 
+You can integrate your CRM platform with Voucherify.
+
+ -->
+
+#### Upserting customers
+
+If the customer does not exist in Voucherify, they are created automatically in a request that includes the customer object. If the customer exists, their data will be upserted. In this case, the response also includes the `updated_at` object. <!-- these are objects, right? -->
+
+The following actions upsert customer data:
+- [Order creation](ref:create-order)
+- [Stackable discount redemption](ref:redeem-stacked-discounts)
+- [Stackable discount redemption (client side)](ref:redeem-stacked-discounts-client-side)
```

Comment:
We should also add client-side publications if we provide the client side stackable redemption

Reply:
Solved on Slack; no change made